sql/memo/exec_builder.go

```diff
@@ -133,6 +133,53 @@ func (b *ExecBuilder) buildLookupJoin(j *LookupJoin, input sql.Schema, children
 	return plan.NewJoin(left, right, j.Op, filters).WithScopeLen(j.g.m.scopeLen), nil
 }
 
+func (b *ExecBuilder) buildSlidingRange(sr *SlidingRange, input sql.Schema, children ...sql.Node) (sql.Node, error) {
+	var ret sql.Node
+	var err error
+
+	leftSch := input[:len(input)-len(sr.Parent.Right.RelProps.OutputCols())]
+	rightSch := input[len(sr.Parent.Left.RelProps.OutputCols()):]
+
+	switch n := children[0].(type) {
```

Comment:
i think the children for this join might be limited to `ResolvedTable` and `TableAlias`. Any PROJECT node can only be pushed into the table. LIMIT and DISTINCT has to wrap the relational expressions (higher in tree). Filter is possible. I might be wrong but I expected the children to be two `IndexedTableAccess` nodes, similar to merge join. It seems like you've noticed this, but the schemas that lookup expressions, join filters, and table filters see will be subtlety different.

Reply:
This is the same children types that `plan.Lookup` checks for.

I refactored slightly, see the new PR for details.